Clarify usage of Distributed Tracing Extension by OpenTelemetry

joaopgrassi opened the pull request:
From the last messaging workgroup meeting, it was pointed out that now that we are close to merge the [CloudEvents Semantic Conventions PR](https://github.com/open-telemetry/opentelemetry-specification/pull/1978), we should make it clear in the CloudEvents spec that the Distributed Tracing Extension is used by OTel. 

The reason for this "linking" between specs is that the extension caused confusion in the past and was even asked to be removed: https://github.com/cloudevents/spec/pull/751. Basically the idea is that OpenTelemetry "claims" this extension and by providing the semantic conventions, its usage becomes more clear. 

Here is the recording of the meeting where this subject came up: https://youtu.be/gjz3Fz_hz_8?t=937

So, in this PR I intended to:

- Remove instructions and extra clarifications on how this is used - It's now in the semantic conventions
- Clear some wording (context, propagation, distributed tracing) by using the text from the OpenTelemetry specification. (For the case people aren't aware of OpenTelemetry nor heard these terms before)
- Add link to the semantic conventions document where users will find out how to use it

joaopgrassi committed:
Add OTel semconv link

joaopgrassi committed:
Add link to CloudEvents semconv

Still not merged, but at least it's there.

joaopgrassi committed:
Fix markdown lint issues

pirgeo commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
-meant to contain historical data of the parent trace, in order to diagnose
-eventual failures of the system through tracing platforms like Jaeger, Zipkin,
-etc.
+[W3C TraceContext](https://www.w3.org/TR/trace-context/) into a CloudEvent.
```
Maybe add a link to https://cloudevents.io/?

joaopgrassi replied:
Hum it's already the CloudEvents specification, so I guess it's clear enough? 😅

pirgeo reviewed (approved):
Looking good to me. I guess this will only be merged after the spec PR, right? the link is currently dead because the `cloudevents.md` is not yet merged.

arminru commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
-etc.
+[W3C TraceContext](https://www.w3.org/TR/trace-context/) into a CloudEvent.
+
+The [OpenTelemetry specification](
```
You could introduce OTel in one sentence with a link to https://opentelemetry.io.

joaopgrassi replied:
I changed it a bit, to try to say how this extension relates to OTel and what OTel is.

arminru commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
+ concerns access their data in-process using the same shared Context object.
+
+Context propagation enables `Distributed Tracing`.
+Also from [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#tracing-signal):
```
```suggestion
Also from the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/overview.md#tracing-signal):
```

arminru commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
+[W3C TraceContext](https://www.w3.org/TR/trace-context/) into a CloudEvent.
+
+The [OpenTelemetry specification](
+https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/context.md#overview)
```
```suggestion
https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/context/context.md#overview)
```
Would make sure the link doesn't go stale as content might be shuffled around.

arminru commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
+## Using the Distributed Tracing Extension
+
+The
+[OpenTelemetry Semantic Conventions for CloudEvents](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/cloudevents.md)
```
```suggestion
[OpenTelemetry Semantic Conventions for CloudEvents](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/trace/semantic_conventions/cloudevents.md)
```
(once released)

arminru commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,43 @@
+
+The
+[OpenTelemetry Semantic Conventions for CloudEvents](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/cloudevents.md)
+defines in which scenarios this extension should be used and how to use it.
```
```suggestion
define in which scenarios this extension should be used and how to use it.
```

joaopgrassi committed:
PR suggestions

Changed also the introduction a bit to "link" this extension with OTel more clearly.

joaopgrassi committed:
Reword things a bit more

pirgeo commented on extensions/distributed-tracing.md:
```diff
@@ -1,55 +1,49 @@
+analyze your software’s performance and behavior.
+
+The OpenTelemetry specification defines both
+[Context](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/context/context.md#overview)
```
Did you add the link to a certain version (`1.8.0`) on purpose? Maybe we should point to latest?

joaopgrassi replied:
Yes, from @arminru's comments. This is to ensure if things move around the links here don't break.
